### ISSUE TITLE
Possible to place objects “inside” other objects (slightly)

### ISSUE THREAD

**Issue by Wuzzy2** [kaa-ching/tbe#131]
It is possible to place objects “inside” other objects so that they end up pushing them around when you start the level. This effect is somewhat subtle, but it allows for cheating.

This issue makes it possible to cheat in the level “float balloon, float” (and possibly other levels, too).
You can cheat by placing a bed of nails just below the right side of the birch bar, so that the left “screw” (?) of the bed of nails is directly below the right end of the birch bar. Start the level and the birch bar is pushed upwards. This is actually how I have “solved” the level on my first try. ;-)
(Later I found out the “real” solution.)

If you activate DebugDrawing, you can (probably) see why, it seems to have something to do with the collision polygon. My guess is that the collision detection in-game is not the same as when you are placing items.

This is not an issue with this particular level, but this level illustrates the problem very well.

c1e3af453fc129d4f132e6b30aa951ad4cf8dfb3

**Comment by kaa-ching** [comment]
This is true, dragging objects around is done on alpha level of the image.

Sensors (like bed of nails and cactus have) do not exist during dragging around.

Not sure if I want to fix this.

**Comment by kaa-ching** [comment]
Indeed, the collision polygon during moving is different - it is based on the outline of the image, whereas during simulation is it based on the model outline of the object.

Due to the new QML UI, I'll have to merge the collision detection for both cases to use the Box2D collision detection anyway. So this issue *should* go away automagically somewhere during milestone C.
